### ISSUE TITLE
Deadline in the issue page revisited

### ISSUE THREAD

**Issue by alexanmtz** [worknenjoy/gitpay#1132]
Change the current deadline feature to make it open using our Drawer component. 

It should have the option to clear the deadline.

Here's the existing one:

![image](https://github.com/user-attachments/assets/7d74279b-c60c-4d9e-84ff-3f376101596c)

We should use the same component but open it in a drawer instead.

And the Task component is too big, separate the component from Task component.

**Comment by RemyaManoharan** [comment]
I would like to work on this task.

**Comment by alexanmtz** [comment]
You're assigned now @RemyaManoharan

**Comment by alexanmtz** [comment]
https://gitpay.me/#/profile/task/961/deadline-in-the-issue-page-revisited

Added a bounty of $43.2 for this one @RemyaManoharan

**Comment by hemangsk** [comment]
@alexanmtz , I'd like to work on this. Looks like the other PR is abandoned.

**Comment by alexanmtz** [comment]
Yes @hemangsk , feel free to proceed with this one and send a PR 🙏

**Comment by hemangsk** [comment]
@alexanmtz Thank you! I've pushed my changes for your review. I checked the CI logs and the failures seems unrelated. It appears to be happening due to rate limit getting exceeded for the github API.

**Comment by alexanmtz** [comment]
Great job @hemangsk 🙏

**Comment by alexanmtz** [comment]
You can claim your bounty here @hemangsk 👌 

https://gitpay.me/#/profile/task/961/deadline-in-the-issue-page-revisited
